src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -1212,45 +1212,165 @@ bool ViewProviderSketch::mouseButtonPressed(int Button, bool pressed, const SbVe
                     return true;
                 case STATUS_NONE: {
                     // A right click shouldn't change the Edit Mode
-                    if (preselection.isPreselectPointValid()) {
-                        return true;
+                    bool noSelection = true;
+                    bool selectPoint = false;
+                    bool selectLine = false;
+                    bool selectConstraint = false;
+                    bool selectMultiple = false;
+
+                    Gui::MenuItem geom;
+                    geom.setCommand("Sketcher context");
+                    // Gets a selection vector
+                    std::vector<Gui::SelectionObject> selection = Gui::Selection().getSelectionEx();
```

Comment:
Here you should use : 
`    std::vector<Gui::SelectionObject> selection = Gui::Selection().getSelectionEx(0, Sketcher::SketchObject::getClassTypeId());`
To make sure that you are selecting only sketches. As it stands I guess that if you select a solid edge from outside the sketch, you get the special menu for edges.